extensions/ecc/circuit/src/ecc_extension.rs

```diff
@@ -44,44 +52,73 @@ pub struct CurveConfig {
     pub b: BigUint,
 }
 
-pub static SECP256K1_CONFIG: Lazy<CurveConfig> = Lazy::new(|| CurveConfig {
+#[serde_as]
+#[derive(Clone, Debug, derive_new::new, Serialize, Deserialize)]
+pub struct TeCurveCoeffs {
+    /// The coefficient a of ax^2 + y^2 = 1 + dx^2y^2
+    #[serde_as(as = "DisplayFromStr")]
+    pub a: BigUint,
+    /// The coefficient d of ax^2 + y^2 = 1 + dx^2y^2
+    #[serde_as(as = "DisplayFromStr")]
+    pub d: BigUint,
+}
+
+pub static SECP256K1_CONFIG: Lazy<CurveConfig<SwCurveCoeffs>> = Lazy::new(|| CurveConfig {
     modulus: SECP256K1_MODULUS.clone(),
     scalar: SECP256K1_ORDER.clone(),
-    a: BigUint::zero(),
-    b: BigUint::from_u8(7u8).unwrap(),
+    coeffs: SwCurveCoeffs {
+        a: BigUint::zero(),
+        b: BigUint::from_u8(7u8).unwrap(),
+    },
 });
 
-pub static P256_CONFIG: Lazy<CurveConfig> = Lazy::new(|| CurveConfig {
+pub static P256_CONFIG: Lazy<CurveConfig<SwCurveCoeffs>> = Lazy::new(|| CurveConfig {
     modulus: P256_MODULUS.clone(),
     scalar: P256_ORDER.clone(),
-    a: BigUint::from_bytes_le(P256_A.as_le_bytes()),
-    b: BigUint::from_bytes_le(P256_B.as_le_bytes()),
+    coeffs: SwCurveCoeffs {
+        a: BigUint::from_bytes_le(P256_A.as_le_bytes()),
+        b: BigUint::from_bytes_le(P256_B.as_le_bytes()),
+    },
+});
+
+pub static ED25519_CONFIG: Lazy<CurveConfig<TeCurveCoeffs>> = Lazy::new(|| CurveConfig {
+    modulus: ED25519_MODULUS.clone(),
+    scalar: ED25519_ORDER.clone(),
+    coeffs: TeCurveCoeffs {
+        a: BigUint::from_bytes_le(ED25519_A.as_le_bytes()),
+        d: BigUint::from_bytes_le(ED25519_D.as_le_bytes()),
+    },
 });
 
 #[derive(Clone, Debug, derive_new::new, Serialize, Deserialize)]
-pub struct WeierstrassExtension {
-    pub supported_curves: Vec<CurveConfig>,
+pub struct EccExtension {
```

Comment:
I can see it's easier to say "ECC Extension" but here I feel like we could have done `WeierstrassExtension` and `TwistedEdwardsExtension` as two separate structs. I don't have strong feelings either way